x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/CrossClusterAccessAuthenticationService.java

```diff
@@ -108,6 +111,71 @@ public void authenticate(final String action, final TransportRequest request, fi
         }
     }
 
+    public void tryAuthenticateCredentialsHeader(Map<String, String> headers, ActionListener<Void> listener) {
+        doTryAuthenticateCredentialsHeader(clusterService.threadPool().getThreadContext(), headers, listener);
+    }
+
+    // visibility for testing
+    public void doTryAuthenticateCredentialsHeader(
+        ThreadContext threadContext,
+        Map<String, String> headers,
+        ActionListener<Void> listener
+    ) {
+        final ApiKeyService.ApiKeyCredentials apiKeyCredentials;
+        try {
+            apiKeyCredentials = getApiKeyCredentialsFromHeaders(headers);
+        } catch (Exception ex) {
+            listener.onFailure(Exceptions.authenticationError("failed to parse cross cluster credentials header", ex));
+            return;
+        }
+
+        apiKeyService.tryAuthenticate(threadContext, apiKeyCredentials, ActionListener.wrap(authResult -> {
+            if (authResult.isAuthenticated()) {
```

Comment:
The auth result and exception handling code is largely adopted from [ApiKeyAuthenticator](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyAuthenticator.java#L58-L78)